code/game/machinery/firealarm.dm

```diff
@@ -53,6 +53,7 @@
 			set_light(l_range = L_WALLMOUNT_HI_RANGE, l_power = L_WALLMOUNT_HI_POWER, l_color = COLOR_RED)
 		else
 			icon_state = "fire0"
+			set_light(0)
```

Comment:
I couldn't replicate the bug to test this. Can you tell me how you replicated it?

Also this fix seems odd to me, since you will be calling set_light anyway after the switch statement below.

Reply:
1. Turn on fire alarm (Alarm + Lights turn red)
2. Turn off fire alarm. (Only lights revert)

I know the lighting subsystems for fire alarms was a bit jank and this fixed the issue. Can't really say more without going back into it.

Reply:
I tried it on a few rooms on the live server. This is the fire alarm after the alarm was reset. It seems correct?

![image_2022-05-05_225415891](https://user-images.githubusercontent.com/47489928/167031872-1802b6ab-4a55-48e5-ab44-23668400abb6.png)

Reply:
I'm able to recreate it consistently and it's been fixed so. *shrug
![image](https://user-images.githubusercontent.com/54920261/167042317-53f3949f-7e9b-42d7-ad99-4744247c1377.png)

Reply:
Just got it live.

![image](https://user-images.githubusercontent.com/54920261/167042400-8704d580-dce6-4bfe-8d89-7c3d2f916b98.png)